azbus/receiver.go

```diff
@@ -199,10 +215,38 @@ func (r *Receiver) elapsed(ctx context.Context, count int, total int, maxDuratio
 	return err
 }
 
-// RenewMessageLock renews the given messages peek lock, so it doesn't lose the lock and get re-added to the message queue.
+// elapsed emits 2 log messages detailing how long processing took. Additionally executes suitable disposition of message.
+func (r *Receiver) elapsedParallel(ctx context.Context, count int, maxDuration time.Duration, msg *ReceivedMessage, handler PHandler) {
+	now := time.Now()
+
+	log := r.log.FromContext(ctx)
+	defer log.Close()
+
+	log.Debugf("Processing message %d", count)
+	disp, ctx, err := r.handleParallelReceivedMessageWithTracingContext(ctx, msg, handler)
```

Comment:
minor: this is excellently named function, but a bit of a mouthful, 

maybe we could use the options pattern or similar to make it less of a mouthful?

```
r.handleParallelMessage(msg, handler, WithTracing(ctx))
```

just a minor though, so can ignore, but happy for other suggestions to reduce the size of this method name :)

Reply:
it is well named :P but gave me java flashbacks!